### DISCUSSION TITLE
Repository and Unit of work pattern in ASP.net core

### DISCUSSION THREAD

**Discussion by giscus** [PradeepLoganathan/pradeepl-blog#22] [BlogPosts]
# Repository and Unit of work pattern in ASP.net core

A Repository is used to manage aggregate persistence and retrieval. The repository mediates between the data-access layer and the domain.The unit of work pattern keeps track of all changes to aggregates. Once all updates of the aggregates in a scope are completed, the tracked changes are played onto the database in a transaction

https://pradeepl.com/blog/repository-and-unit-of-work-pattern-asp-net-core-3-1/

**Comment by macoratti** [comment]
Hi,
I'm learning about Unit Of Work and in your implementation you are injecting the repository instances in the Unit Of Work's constructor, but it doesn't force my application to always use these instances even though I don't need it ?
Thanks
